Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerInitialize.rst.txt

```diff
@@ -0,0 +1,46 @@
+.. Generated by https://github.com/TYPO3-Documentation/t3docs-codesnippets 
+.. Extracted from T3docs\Examples\LinkHandler\GitHubLinkHandler
+
+.. code-block:: php
+   :caption: Class T3docs\\Examples\\LinkHandler\\GitHubLinkHandler
+
+   use TYPO3\CMS\Backend\Controller\AbstractLinkBrowserController;
+   use TYPO3\CMS\Backend\View\BackendViewFactory;
+   use TYPO3\CMS\Core\Page\PageRenderer;
+   use TYPO3\CMS\Core\Utility\GeneralUtility;
+   use TYPO3\CMS\Core\View\ViewInterface;
+   
+   class GitHubLinkHandler implements LinkHandlerInterface
+   {
+       protected array $linkAttributes = ['target', 'title', 'class', 'params', 'rel'];
+   
+       protected array $linkParts = [];
+   
+       protected AbstractLinkBrowserController $linkBrowser;
+   
+       protected ViewInterface $view;
+   
+       protected PageRenderer $pageRenderer;
+   
+       protected array $configuration;
+   
+       public function __construct(
+           protected readonly BackendViewFactory $backendViewFactory,
```

Comment:
What for is the BackendViewFactory?

Reply:
Might be a relict, however we need to anylize and fix this in examples not here

Reply:
Yup, that should be done there before merging here.